### PR TITLE
Springboot security

### PR DESCRIPTION
**Pull request by qiaoxibj**
Add new Springboot Jersey and Springboot Security project

---
This change is [Reviewable](https://reviewable.io/reviews/wasdev/tool.accelerate.core/41)

### CODE REVIEW

File: starter-microservice-springboot-jersey/src/main/webapp/WEB-INF/classes/description.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<div>
+	<h2>Spring Boot with Jersey</h2>
+	<p>Inside the application project there is a application.springboot.jersey package
+		containing 6 classes:</p>
```
Comment (iainduncani) [inline starter-microservice-springboot-jersey/src/main/webapp/WEB-INF/classes/description.html]: Is that supposed to be 3 classes or are some not listed?

File: starter-microservice-springboot-jersey/src/main/webapp/sample/myProject-application/src/main/java/application/springboot/jersey/Endpoint.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import javax.ws.rs.core.MediaType;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Component
+@Path("/")
+public class Endpoint {
```
Comment (iainduncani) [inline starter-microservice-springboot-jersey/src/main/webapp/sample/myProject-application/src/main/java/application/springboot/jersey/Endpoint.java]: Can we call this JerseyEndpoint or RestEndpoint instead?

File: starter-microservice-springboot-security/src/main/webapp/sample/myProject-application/src/main/java/application/springboot/security/WebSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            .formLogin()
+                .loginPage("/springbootsecurity/login")
+                .permitAll()
+                .and()
+            .logout()
+                .permitAll();        
+    }
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
```
Comment (iainduncani) [inline starter-microservice-springboot-security/src/main/webapp/sample/myProject-application/src/main/java/application/springboot/security/WebSecurityConfig.java]: Do you know of anyway to get the user information from a simple user registry in the server.xml instead?

File: starter-microservice-springboot-security/src/test/java/com/ibm/liberty/starter/service/springboot/security/api/v1/it/EndpointTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	       return (T) mapper.readValue(json, entity);       	
+        } finally {
+        	if(response != null) {
+        		response.close();
+        	}
+        }
+    }
+
+    public Response sendRequest(String url, String requestType) {
+        Client client = ClientBuilder.newClient();
```
Comment (iainduncani) [inline starter-microservice-springboot-security/src/test/java/com/ibm/liberty/starter/service/springboot/security/api/v1/it/EndpointTest.java]: I don't see this adding the security information,how does it work?

File: starter-microservice-springboot-security/src/main/webapp/WEB-INF/classes/description.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<div>
+	<h2>Spring Boot with Security</h2>
+	<p>Inside the application project there is a application.springboot.web package
+		containing two classes:</p>
```
Comment (katheris) [inline starter-microservice-springboot-security/src/main/webapp/WEB-INF/classes/description.html]: Should this say three classes?

File: README.md [hunk 6]
```diff
@@ -30,6 +30,24 @@ Inside the application project there is a application.springboot.web package con
 
 Inside the wlpcfg project there is the <code>it.springboot.web.HelloControllerTest</code> that will test the Spring MVC endpoint to ensure it is working.
 
+## Spring Boot with Jersey
+This provides you with a SpringBoot application that will run on WebSphere Liberty.
+
+Inside the application project there is a application.springboot.jersey package containing 6 classes:
```
Comment (katheris) [inline README.md]: There aren't 6 classes listed here

File: starter-microservice-springboot-jersey/src/main/webapp/WEB-INF/classes/description.html (hunk 1 above)
Comment (qiaoxibj) [inline starter-microservice-springboot-jersey/src/main/webapp/WEB-INF/classes/description.html]: typo fixed in new commit

File: starter-microservice-springboot-security/src/main/webapp/sample/myProject-application/src/main/java/application/springboot/security/WebSecurityConfig.java (hunk 3 above)
Comment (qiaoxibj) [inline starter-microservice-springboot-security/src/main/webapp/sample/myProject-application/src/main/java/application/springboot/security/WebSecurityConfig.java]: @iainduncani , I dont think it is reasonable to mix springboot security and liberty security, what do you think?

File: starter-microservice-springboot-security/src/test/java/com/ibm/liberty/starter/service/springboot/security/api/v1/it/EndpointTest.java (hunk 4 above)
Comment (qiaoxibj) [inline starter-microservice-springboot-security/src/test/java/com/ibm/liberty/starter/service/springboot/security/api/v1/it/EndpointTest.java]: this is only test the security entry page can be access, not contain the testing for protected page.

**Pull request closed without merging**